yb-voyager/cmd/exportSchema.go

```diff
@@ -195,6 +209,84 @@ func exportSchema() error {
 	return nil
 }
 
+func runAssessMigrationCmdBeforExportSchemaIfRequired(exportSchemaCmd *cobra.Command) error {
+	if source.DBType != POSTGRESQL {
+		log.Infof("Skipping running assess-migration command from export schema as source DB type is not PostgreSQL.")
+		return nil
+	}
+
+	if ok, _ := IsMigrationAssessmentDoneDirectly(metaDB); ok {
+		log.Infof("Migration assessment is already done, skipping running assess-migration command.")
+		return nil
+	} else if ok, _ := IsMigrationAssessmentDoneViaExportSchema(); ok {
+		log.Infof("Migration assessment is already done via export schema, skipping running assess-migration command.")
+		return nil
+	}
+
+	// Temporarily override utils.ErrExit with panic so it doesn’t kill the export schema command.
+	utils.SetExitHook(func(code int) { panic(fmt.Sprintf("exit %d", code)) })
+	defer utils.SetExitHook(nil) // restore when we’re done
+
+	// Note: have to maintain it in metaDB, can't rely on in-memory var to cover multiple runs scenario
+	isAssessmentInvokedFromExportSchema = true
+
+	var assessFlagsWithValues []string
+	commonFlags := utils.GetCommonFlags(exportSchemaCmd, assessMigrationCmd)
+	for _, flag := range commonFlags {
+		// don't pass start-clean flag to assess-migration command here
+		if flag.Name == "start-clean" || !flag.Changed {
+			continue
+		}
+
+		// bool flags: --flag=value
+		// everything else: --flag value
+		if flag.Value.Type() == "bool" {
+			assessFlagsWithValues = append(assessFlagsWithValues,
+				fmt.Sprintf("--%s=%s", flag.Name, flag.Value.String()),
+			)
+		} else {
+			assessFlagsWithValues = append(assessFlagsWithValues,
+				"--"+flag.Name,
+				flag.Value.String(),
+			)
+		}
+	}
+
+	// Append --yes=true at the end to override any --yes=false if set in export schema cmd
+	assessFlagsWithValues = append(assessFlagsWithValues,
+		"--iops-capture-interval", "0", // TODO: any small but significant duration will be better than 0
+		"--target-db-version", ybversion.LatestStable.String(),
+		"--yes",
+	)
+
+	utils.PrintAndLog("\n[debug] Running assessment before export schema with flags: %s\n", strings.Join(assessFlagsWithValues, "\t"))
+
+	// Silence any built-in error; will be handled by the caller(exportSchemaCmd)
+	assessMigrationCmd.SilenceErrors = true
+
+	// bring in all persistent flags(--export-dir, --yes) so Parse() can recognize them
+	assessMigrationCmd.Flags().AddFlagSet(assessMigrationCmd.PersistentFlags())
```

Comment:
this is a bit weird. Do you know - Why aren't the flags already part of the cmd?